app/src/main/java/com/github/se/icebreakrr/MainActivity.kt

```diff
@@ -36,24 +40,62 @@ import com.github.se.icebreakrr.ui.sections.FilterScreen
 import com.github.se.icebreakrr.ui.sections.NotificationScreen
 import com.github.se.icebreakrr.ui.sections.SettingsScreen
 import com.github.se.icebreakrr.ui.theme.SampleAppTheme
+import com.github.se.icebreakrr.utils.PermissionManager
+import com.google.android.gms.location.FusedLocationProviderClient
+import com.google.android.gms.location.LocationServices
 import com.google.firebase.FirebaseApp
 import com.google.firebase.auth.FirebaseAuth
 import com.google.firebase.functions.FirebaseFunctions
 
 class MainActivity : ComponentActivity() {
   private lateinit var auth: FirebaseAuth
   private lateinit var functions: FirebaseFunctions
+  private lateinit var locationViewModel: LocationViewModel
+  private lateinit var locationService: LocationService
+  private lateinit var geoFirestoreRepository: GeoFirestoreRepository
+  private lateinit var permissionManager: PermissionManager
+  private lateinit var authStateListener: FirebaseAuth.AuthStateListener
+  private lateinit var fusedLocationClient: FusedLocationProviderClient
 
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
+
     // Retrieve the testing flag from the Intent
     val isTesting = intent?.getBooleanExtra("IS_TESTING", false) ?: false
 
-    requestNotificationPermission()
+    // Create and initialize the PermissionManager with the list of permissions required
+    requestNotificationPermission() // TODO remove this and use the PermissionManager instead
```

Comment:
`requestNotificationPermission() // TODO remove this and use the PermissionManager instead`
Question: just to be sure, you commented that for Maximo to do it later, right?

Reply:
Yup exactly ;)